pkg/agent/util/dump.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"github.com/spf13/afero"
+	"k8s.io/utils/exec"
+
+	"github.com/vmware-tanzu/antrea/pkg/ovs/ovsctl"
+)
+
+// Dumper is the interface for dumping logs of the controller. Its functions should
+// only work in agent Pod or a Windows Node which installed an agent.
+type Dumper interface {
+	DumpFlows(basedir string) error
+	DumpHostNetworkInfo(basedir string) error
+	DumpLog(basedir string) error
+}
+
+type ofctlClientFactory func(bridge string) ovsctl.OVSCtlClient
+
+func NewDumper(fs afero.Fs, executor exec.Interface, factory ofctlClientFactory) Dumper {
+	return &dumper{
```

Comment:
why duplicate struct dumper are defined in platform specific code?

Reply:
Thanks, removed.